src/components/Payroll/PayrollOverview/PayrollOverviewPresentation.tsx

```diff
@@ -1,74 +1,470 @@
+              ),
+            },
+            {
+              title: t('tableHeaders.employeeTaxes'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>
+                  {formatCurrency(
+                    employeeCompensations.taxes?.reduce(
+                      (acc, tax) => (tax.employer ? acc : acc + tax.amount),
+                      0,
+                    ) ?? 0,
+                  )}
+                </Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.employeeBenefits'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>
+                  {formatCurrency(
+                    employeeCompensations.benefits?.reduce(
+                      (acc, benefit) => acc + (benefit.employeeDeduction ?? 0),
+                      0,
+                    ) ?? 0,
+                  )}
+                </Text>
+              ),
+            },
+            {
+              title: t('tableHeaders.payment'),
+              render: (employeeCompensations: EmployeeCompensations) => (
+                <Text>{formatCurrency(employeeCompensations.netPay ?? 0)}</Text>
+              ),
+            },
+          ]}
+          data={payrollData.employeeCompensations!}
+        />
+      ),
+    },
+  ]
 
   return (
     <Flex flexDirection="column" alignItems="stretch">
       <Flex justifyContent="space-between">
-        <Heading as="h1">{t('pageTitle', { startDate: 'Jul 5', endDate: 'Jul 18, 2025' })}</Heading>
+        <div>
+          <Heading as="h1">{t('pageTitle')}</Heading>
+          <Text>{getPayrollOverviewTitle({ payPeriod: payrollData.payPeriod, locale, t })}</Text>
+        </div>
         <Flex justifyContent="flex-end">
-          <Button title={t('buttons.editTitle')} onClick={onEdit} variant="secondary">
-            {t('buttons.edit')}
-          </Button>
-          <Button title={t('buttons.submitTitle')} onClick={onSubmit}>
-            {t('buttons.submit')}
+          <Button onClick={onEdit} variant="secondary">
+            {t('editCta')}
           </Button>
+          <Button onClick={onSubmit}>{t('submitCta')}</Button>
         </Flex>
       </Flex>
+      {/* TODO: when is this actually saved? */}
       <Alert label={t('alerts.progressSaved')} status="success"></Alert>
       <Alert
         label={t('alerts.directDepositDeadline', {
-          payDate: 'Fri, Jul 25',
-          deadline: '7:00 PM EDT on Wed, Jul 23',
+          payDate: parseDateStringToLocal(payrollData.checkDate!)?.toLocaleDateString(locale, {
+            month: 'long',
+            day: 'numeric',
+          }),
+          deadline: payrollData.payrollDeadline?.toLocaleString(locale, {
+            month: 'long',
+            day: 'numeric',
+            year: 'numeric',
+            hour: '2-digit',
+            minute: '2-digit',
+            hour12: true,
+          }),
```

Comment:
Based on what i'm seeing in the design, this looks like it actually belongs over in the PayrollConfiguration component. Ex. see PayrollOverview here without the alert https://www.figma.com/design/6dxOSiONDiJoa9zY1wOwbs/Frontend-SDK-Partner-Design-File?node-id=9604-28976&t=n8Ys6vSxrlJ3BUOp-4 and PayrollConfiguration featuring this here https://www.figma.com/design/6dxOSiONDiJoa9zY1wOwbs/Frontend-SDK-Partner-Design-File?node-id=9823-18658&t=n8Ys6vSxrlJ3BUOp-4